src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToSubmitHandler.java

```diff
@@ -51,7 +51,9 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
         // on the final submission from CCD, so we need to reset it here
         // See https://tools.hmcts.net/jira/browse/RDM-8200
         // This is a temporary workaround for this issue.
-        sscsCaseData.setWriteFinalDecisionGeneratedDate(LocalDate.now().toString());
+        if (sscsCaseData.getWriteFinalDecisionGeneratedDate() == null) {
```

Comment:
Im not sure if this is going to work for the following scenario:
- Final decision generated on say 1st August. Document generated date says '1st August' and the writeFinalDecisionGeneratedDate field is saved with '1st August'
- Peer reviewed offline and they need to make a change the following day. The document generated date for the preview says '2nd August'. 
- The Final decision is then submitted. Due to line 54 only overwriting the writeFinalDecisionGeneratedDate if it is null, then it will leave it as the 1st August (even though the preview says the 2nd August. 
- When the document is issued, it will use the date that is saved in CCD which is the 1st August - this is not correct.

Reply:
Thanks for going through this Jack.   We've now demonstrated that this is indeed the case - I've removed the conditional null check